crates/bindings-csharp/Runtime/Runtime.cs

```diff
@@ -61,18 +61,21 @@ public AlgebraicType GetAlgebraicType(ITypeRegistrar registrar) =>
 public abstract record BaseReducerContext<DbView> : DbContext<DbView>, IReducerContext
     where DbView : class, new()
 {
-    public Identity Sender => Runtime.SenderIdentity!;
-    public Address Address => Runtime.SenderAddress!;
-    public DateTimeOffset Time => Runtime.Time!;
-}
-
-public static class Runtime
-{
-    public static Random Random { get; internal set; } = new();
-
-    public static Identity? SenderIdentity { get; internal set; }
+    public readonly Identity Sender;
+    public readonly Address? Address;
+    public readonly Random Random;
+    public readonly DateTimeOffset Time;
 
-    public static Address? SenderAddress { get; internal set; }
-
-    public static DateTimeOffset Time { get; internal set; }
+    protected BaseReducerContext(
```

Comment:
Looks like this can be converted into shorthand record syntax now?

Reply:
At that point I'm actually not even sure that we need `BaseReducerContext`, as `ReducerContext` just forwards all arguments to it.

Could we make `ReducerContext` inherit from `DbContext` on its own?

We have `IReducerContext` interface for erased types anyway.